pkg/lib/cockpit.js

```diff
@@ -1483,10 +1483,24 @@ function factory() {
             });
     };
 
-    function format_units(number, suffixes, factor, options) {
-        // backwards compat: "options" argument position used to be a boolean flag "separate"
-        if (!is_object(options))
-            options = { separate: options };
+    let deprecated_format_warned = false;
+    function format_units(suffixes, number, second_arg, third_arg) {
+        let options = second_arg;
+        let factor = options?.base2 ? 1024 : 1000;
+
+        // compat API: we used to accept 'factor' as a separate second arg
+        if (third_arg || (second_arg && !is_object(second_arg))) {
+            if (!deprecated_format_warned) {
+                console.warn(`cockpit.format_{bytes,bits}[_per_sec](..., ${second_arg}, ${third_arg}) is deprecated.`);
```

Comment:
When do we remove it, should we include that? Like in 10 releases this API is gone?